runner/clients/proxy/proxy.go

```diff
@@ -10,30 +10,37 @@ package proxy
 
 import (
 	"bytes"
+	"compress/gzip"
 	"context"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
 
 	"github.com/base/base-bench/runner/clients/types"
 	"github.com/ethereum-optimism/optimism/op-service/client"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
 )
 
 type ProxyServer struct {
-	client types.ExecutionClient
-	log    log.Logger
-	port   int
-	server *http.Server
+	client     types.ExecutionClient
+	log        log.Logger
+	port       int
+	server     *http.Server
+	pendingTxs []*ethTypes.Transaction
+	nonce      int
 }
 
 func NewProxyServer(client types.ExecutionClient, log log.Logger, port int) *ProxyServer {
 	return &ProxyServer{
 		client: client,
 		log:    log,
 		port:   port,
+		nonce:  0,
```

Comment:
Would be nice if the mempool tracked this instead. Also, I think we'd want this per sender address rather than a global value.